glue-codes/openfast-cpp/src/OpenFAST.cpp

```diff
@@ -1291,7 +1291,10 @@ void fast::OpenFAST::advance_to_next_driver_time_step(bool writeFiles) {
           int tStepRatio = time_step_ratio(dtFAST, dtDriver);
           if ( (restartFreq_*tStepRatio > 0) && (((nt_global - ntStart) % (restartFreq_*tStepRatio)) == 0 )  && (nt_global != ntStart) ) {
               turbineData[iTurb].FASTRestartFileName = " "; // if blank, it will use FAST convention <RootName>.nt_global
-              FAST_CreateCheckpoint(&iTurb, turbineData[iTurb].FASTRestartFileName.data(), &ErrStat, ErrMsg);
+              char tmpRstFileRoot[INTERFACE_STRING_LENGTH];
+              strncpy(tmpRstFileRoot, turbineData[iTurb].FASTRestartFileName.c_str(), turbineData[iTurb].FASTRestartFileName.size());
+              tmpRstFileRoot[turbineData[iTurb].FASTRestartFileName.size()] = '\0';
+              FAST_CreateCheckpoint(&iTurb, tmpRstFileRoot, &ErrStat, ErrMsg);
```

Comment:
Would you consider trying the following to avoid the C-style string handling? It should work the same.

```suggestion
              std::string tmpRstFileRoot{turbineData[iTurb].FASTRestartFileName};
              tmpRstFileRoot.resize(INTERFACE_STRING_LENGTH, ' ');
              FAST_CreateCheckpoint(&iTurb, tmpRstFileRoot.c_str(), &ErrStat, ErrMsg);
```

Reply:
I am happy to try that. My current implementation follows what is already done in this file, e.g. on L638 (https://github.com/OpenFAST/openfast/blob/main/glue-codes/openfast-cpp/src/OpenFAST.cpp#L638).  Would you want me to change that instance as well?

Reply:
If you don't mind. Windows doesn't like `strncpy`, and while we don't compile the `OpenFAST.cpp`, it's likely that people will take it as an example and use it elsewhere. Thanks!